src/docker-in-docker/install.sh

```diff
@@ -377,16 +379,24 @@ dockerd_start="$(cat << 'INNEREOF'
         echo "Not setting dockerd DNS manually."
         CUSTOMDNS=""
     fi
+
     set -e
 
+    if [ -z "$DOCKER_DEFAULT_ADDRESS_POOL" ] || [ "$DOCKER_DEFAULT_ADDRESS_POOL" = "" ]
+    then
+        DEFAULT_ADDRESS_POOL=""
+    else
+        DEFAULT_ADDRESS_POOL="--default-address-pool $DOCKER_DEFAULT_ADDRESS_POOL"
+    fi
+
     # Start docker/moby engine
-    ( dockerd $CUSTOMDNS > /tmp/dockerd.log 2>&1 ) &
+    ( dockerd $CUSTOMDNS $DEFAULT_ADDRESS_POOL > /tmp/dockerd.log 2>&1 ) &
 INNEREOF
 )"
 
 # Start using sudo if not invoked as root
 if [ "$(id -u)" -ne 0 ]; then
-    sudo /bin/sh -c "${dockerd_start}"
+    sudo -E /bin/sh -c "${dockerd_start}"
```

Comment:
Passing the local environment to `sudo` shouldn't be necessary.

```suggestion
    sudo /bin/sh -c "${dockerd_start}"
```

Reply:
If `-E` isn't present then the command in `$dockerd_start` doesn't have the values of `AZURE_DNS_AUTO_DETECTION` or `DOCKER_DEFAULT_ADDRESS_POOL`.